baseapp/baseapp.go

```diff
@@ -200,7 +202,7 @@ func (app *BaseApp) setCheckState(header abci.Header) {
 	ms := app.cms.CacheMultiStore()
 	app.checkState = &state{
 		ms:  ms,
-		ctx: sdk.NewContext(ms, header, true, app.Logger),
+		ctx: sdk.NewContext(ms, header, true, app.Logger).WithConsensusParams(app.consensusParams),
```

Comment:
As per the above comment, I think we need this as well for DeliverTx, rather than app.consensusParams.